app/models/chess_piece.rb

```diff
@@ -13,29 +13,31 @@ def self.types
   scope :bishops, -> { where(type: 'Bishop') }
   scope :pawns, -> { where(type: 'Pawn') }
 
-  def move_to!(new_x, new_y)
-    space = find_piece(new_x, new_y)
+  def move_to!(current_location, destination)
+    x1 = current_location[0]
+    y1 = current_location[1]
+    x2 = destination[0]
+    y2 = destination[1]
+    space = find_piece(x2, y2).first
+    puts space
 
+    valid_move?([x1, y1], [x2, y2])
```

Comment:
I only have one question with this line of code when calling the valid_move?. Would this call the method for the type of piece that is being moved? how does the code use the pawn valid_move? vs the bishops valid_move?